frontend/src/components/dialog/EditPeriodDialog.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <dialog-form
+    v-model="showDialog"
+    icon="mdi-calendar-edit"
+    :title="entityData.description"
+    max-width="600px"
+    :submit-action="update"
+    :cancel-action="close">
+    <template v-slot:activator="scope">
+      <slot name="activator" v-bind="scope" />
+    </template>
+
+    <v-row>
+      <v-col cols="12">
+        <v-text-field
+          v-model="entityData.description"
+          label="Description"
+          required />
+      </v-col>
+      <v-col cols="12">
+        <v-text-field
+          v-model="entityData.start"
+          label="Start"
+          required />
+      </v-col>
+      <v-col cols="12">
+        <v-text-field
+          v-model="entityData.end"
+          label="End"
+          required />
+      </v-col>
+    </v-row>
+  </dialog-form>
+</template>
+
+<script>
+import DialogBase from './DialogBase'
+import DialogForm from './DialogForm'
+export default {
+  name: 'EditPeriodDialog',
+  components: { DialogForm },
+  extends: DialogBase,
+  props: {
+    period: { type: Object, required: true }
```

Comment:
@carlobeltrame Findest du das ist korrekt so oder müssten wir `period` besser als `Function` übergeben?

Reply:
Leider geht es momentan eben nicht auf einfache Art konsistent.
* Wenn die Prop von aussen mit einem Listen-Element befüllt wird ist es `Object` (z.B. `v-for="camp().periods().items"`)
* Wenn die Prop von aussen mit einer Property aus der API befüllt wird kann es beides sein, muss man von Fall zu Fall entscheiden (z.B. `camp().creator` oder `camp().creator()`)
* Wenn die Prop aus der Route befüllt wird ist es immer eine Funktion weil Router-Props nicht reaktiv sind (z.B. `campFromRoute`). Siehe https://router.vuejs.org/guide/essentials/passing-props.html#object-mode "Try to keep the props function stateless, as it's only evaluated on route changes."

Reply:
Ah, ok. Hab auch grad gemerkt, dass das nicht so einfach wird. Ich lass mal so wie's jetzt ist